plugins/modules/dcnm_network.py

```diff
@@ -1975,6 +1975,133 @@ def failure(self, resp):
 
         self.module.fail_json(msg=res)
 
+    def dcnm_update_network_information(self, want, have, cfg):
+
+        if cfg.get("vrf_name", None) is None:
+            want["vrf"] = have["vrf"]
+
+        if cfg.get("net_id", None) is None:
+            want["networkId"] = have["networkId"]
+
+        if cfg.get("net_template", None) is None:
+            want["networkTemplate"] = have["networkTemplate"]
+
+        if cfg.get("net_extension_template", None) is None:
+            want["networkExtensionTemplate"] = have["networkExtensionTemplate"]
+
+        json_to_dict_want = json.loads(want['networkTemplateConfig'])
+        json_to_dict_have = json.loads(have['networkTemplateConfig'])
+
+        if cfg.get("vlan_id", None) is None:
+            json_to_dict_want["vlanId"] = json_to_dict_have["vlanId"]
+            if json_to_dict_want["vlanId"] != "":
+                json_to_dict_want["vlanId"] = int(json_to_dict_want["vlanId"])
+
+        if cfg.get("routing_tag", None) is None:
+            json_to_dict_want["tag"] = json_to_dict_have["tag"]
+            if json_to_dict_want["tag"] != "":
+                json_to_dict_want["tag"] = int(json_to_dict_want["tag"])
+
+        if cfg.get("gw_ip_subnet", None) is None:
+            json_to_dict_want["gatewayIpAddress"] = json_to_dict_have["gatewayIpAddress"]
+
+        if cfg.get("is_l2only", None) is None:
+            json_to_dict_want["isLayer2Only"] = json_to_dict_have["isLayer2Only"]
+            if json_to_dict_want["isLayer2Only"] == "true":
```

Comment:
Just to be safe you might want to use the to_lower() function.  That way of the string were to ever change to `"True"` for instance it would not break.

```python
if json_to_dict_want["isLayer2Only"].to_lower() == "true":
```